### PR TITLE
circuit v3 beta.1 changes

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: .mocharc.json [hunk 1]
```diff
@@ -4,5 +4,6 @@
     ],
     "spec": "tests/**/*.test.ts",
     "require": "ts-node/register",
-    "timeout": "240000"
+    "timeout": "240000",
+    "maxDiffSize": "0"
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply .mocharc.json]: Yes, it's braking for linked multy query test with default value of `8192`. Let's setup it to `10000` for now.

File: tests/proofs/sig-onchain.test.ts [hunk 2]
```diff
@@ -185,10 +185,15 @@ describe('sig onchain proofs', () => {
 
     expect(vp).to.be.undefined;
 
-    // query hash test
-    expect(pub_signals[2]).to.be.equal(
-      '15045271939084694661437431358729281571840804299863053791890179002991342242959'
-    );
+    if (circuitId === CircuitId.AtomicQueryV3OnChain) {
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply tests/proofs/sig-onchain.test.ts:188]: not sure, for V3 we have another query hash calculation, so just hardcoded precalculated hashes for unit test here (both V2 and V3)

File: tests/circuits/utils.ts [hunk 3]
```diff
@@ -354,18 +354,56 @@ export function calculateQueryHash(
...
+    BigInt(operator),
+    BigInt(claimPathKey),
+    BigInt(claimPathNotExists),
+    valueHash,
+    BigInt(valueArraySize)
+  ]);
+  return queryHash;
+}
+
+export function calculateQueryHashV3(
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply tests/circuits/utils.ts]: added as part of SDK to utils

File: src/circuits/models.ts [hunk 4]
```diff
@@ -51,6 +51,48 @@ export class Query {
...
+      Operators.GT,
+      Operators.NE,
+      Operators.LTE,
+      Operators.GTE,
+      Operators.EXISTS
+    ];
+    const twoArrSizeOps = [Operators.BETWEEN, Operators.NOTBETWEEN];
+    const maxArrSizeOps = [Operators.IN, Operators.NIN];
+
+    if (oneArrSizeOps.includes(this.operator)) {
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply src/circuits/models.ts:68]: not sure how to do that for includes

File: src/circuits/comparer.ts [hunk 5]
```diff
@@ -30,6 +30,8 @@ export enum Operators {
   LTE = 7,
   GTE = 8,
   BETWEEN = 9,
+  NOTBETWEEN = 10,
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply src/circuits/comparer.ts]: added

File: src/circuits/models.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply src/circuits/models.ts]: done

File: src/circuits/comparer.ts [hunk 6]
```diff
@@ -168,6 +177,16 @@ export class Vector implements IComparer {
         return this.y.includes(this.x);
       case Operators.NIN:
         return !this.y.includes(this.x);
+      case Operators.BETWEEN:
+        if (this.y.length !== 2) {
+          return false;
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply src/circuits/comparer.ts:182]: Not sure, in go implementation we just return false. 
If we want to throw, we would also need to throw for IN/NIN if values length is 0 or more 64. Let's keep it simple here, there are other places that check arr size

**Review by vmidyllic — APPROVED** [review]

File: src/iden3comm/handlers/contract-request.ts [hunk 7]
```diff
@@ -148,7 +148,7 @@ export class ContractRequestHandler implements IContractRequestHandler {
         {
           skipRevocation: query.skipClaimRevocationCheck ?? false,
           challenge: opts.challenge,
-          verifierDid
+          verifier: DID.idFromDID(verifierDid).bigInt()
```
Comment (vmidyllic) [inline src/iden3comm/handlers/contract-request.ts:151]: this is an issue. we must take verifier id from contract address (convert eth address to big int (with bytes swap)

File: src/storage/filters/jsonQuery.ts [hunk 8]
```diff
@@ -22,7 +22,8 @@ export type FilterOperatorMethod =
   | '$ne'
   | '$gte'
   | '$lte'
-  | '$sd';
+  | '$sd'
```
Comment (vmidyllic) [inline src/storage/filters/jsonQuery.ts:25]: do we need to add between / nonbetween operators here

**Review by vmidyllic — CHANGES_REQUESTED** [review]

File: src/storage/filters/jsonQuery.ts (hunk 8 above)
Comment (volodymyr-basiuk) [reply src/storage/filters/jsonQuery.ts:25]: yes, added

File: src/iden3comm/handlers/contract-request.ts (hunk 7 above)
Comment (volodymyr-basiuk) [reply src/iden3comm/handlers/contract-request.ts]: fixed

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**